### PR TITLE
perf: only record return value if required

### PR DESCRIPTION
**Pull request by mattsse**
~~Blocked by https://github.com/bluealloy/revm/pull/509~~

this will only record the call's output if configured

prep for #3032

### CODE REVIEW

**Comment by onbjerg** [comment]
Blocking PR merged, marking as unblocked @mattsse

**Comment by mattsse** [comment]
this is still blocked because revm has two branches and the feature we need here is on main but we pull in release/v25

**Comment by mattsse** [comment]
@onbjerg unblocked by copying the unreleased function to unblock this

File: crates/rpc/rpc/src/eth/revm_utils.rs [hunk 1]
```diff
@@ -524,6 +524,18 @@ where
     }
 }
 
+/// Helper to get the output data from a result
+///
+/// Can be phased out when <https://github.com/bluealloy/revm/pull/509> is released
```
Comment (onbjerg) [inline crates/rpc/rpc/src/eth/revm_utils.rs:529]:
```suggestion
/// TODO: Can be phased out when <https://github.com/bluealloy/revm/pull/509> is released
```

**Review by onbjerg — APPROVED** [review]

**Pull request merged**